utils/interfaces/_library/core.py

```diff
@@ -124,8 +125,9 @@ def add_profiling_validation(self, validator):
     def profiling_assert_field(self, field_name, content_pattern=None):
         self.append_validation(_ProfilingFieldAssertion(field_name, content_pattern))
 
-    def assert_trace_exists(self, request):
-        self.append_validation(_TraceExistence(request=request))
+    def assert_trace_exists(self, request, span_type):
+        self.append_validation(_StatusCodeValidation(request=request, status_code=200))
```

Comment:
It's an implicit test, not a very good design. I'd rather move this line in test file.

Reply:
But what if it's an assertion we'd like to share across an entire category of tests? We then copy paste the same code in every test?

Reply:
for `assert_trace_exists`, I would say yes, because by it's name, it test trace existance, and don't care about the status code. So yes, explicit this one.

But let's say that checking status code would made sense, In that situation, we would modify the  assert_trace_exists signature in that way : 

```
 def assert_trace_exists(self, request, span_type=None, request_status_code=None):
```

if `request_status_code` is not specified => do not check it. If it's specified, check it.

Bu again, for this specific use case, I really feel that explicit is better in term of test code clarity, and furthermore more on test output.

Reply:
I'm okay with explicit for now. I think we can save a lot of effort and pain later by accepting a structure of conventions for assertions.